### PR TITLE
Wizard of Wikipedia MTurk Task

### PR DESCRIPTION
**Pull request by klshuster**
Wizard of Wikipedia Mturk Task

cc @jaseweston

### CODE REVIEW

File: parlai/mturk/tasks/wizard_of_wikipedia/worlds.py [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+                            self.persona_to_topics[prev_p] = [topic]
+                    else:
+                        prev_p = p_i
+
+    def get_topics(self, persona_sent):
+        return self.persona_to_topics.get(persona_sent, [])
+
+
+class RoleOnboardWorld(MTurkOnboardWorld):
+    '''A world that provides
```
Comment (emilydinan) [inline parlai/mturk/tasks/wizard_of_wikipedia/worlds.py]: double quotation marks here

File: parlai/mturk/tasks/wizard_of_wikipedia/worlds.py [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+                     'mtdo': convo_finished and self.good_wiz,
+                     'mtdo_worker': self.wizard_worker if self.good_wiz else '',
+                     'mtdont_worker': self.wizard_worker if not self.good_wiz else ''},
+                    open(filename, 'wb'))
+        print('{}: Data successfully saved at {}.'.format(
+            self.world_tag,
+            filename))
+
+    def mtdo(self):
+        '''Determines whether to soft-block this turker or not
```
Comment (emilydinan) [inline parlai/mturk/tasks/wizard_of_wikipedia/worlds.py]: double quotation marks

File: parlai/mturk/tasks/wizard_of_wikipedia/worlds.py [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+        """Shutdown all mturk agents in parallel, otherwise if one mturk agent
+        is disconnected then it could prevent other mturk agents from
+        completing.
+        """
+        global shutdown_agent
+
+        def shutdown_agent(agent):
+            # try:
+            agent.shutdown()
+            # except Exception:
```
Comment (emilydinan) [inline parlai/mturk/tasks/wizard_of_wikipedia/worlds.py]: can get rid of these comments

**Comment by namanahuja** [comment]
Hi,
This is an interesting dataset. Any suggestions on the model to be used and its hyper-parameters?

**Comment by jaseweston** [comment]
See the paper for models:
https://openreview.net/forum?id=r1l73iRqKm

**Pull request merged**